packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -57,6 +87,33 @@ const ButtonText = styled(BaseText)(
   }),
 );
 
+type ColorTokenConfig = {
+  property: 'background' | 'border' | 'text' | 'icon';
+  variant: Required<BaseButtonProps['variant']>;
+  state: 'default' | 'hover' | 'active' | 'focus' | 'disabled';
+  intent: BaseButtonProps['intent'];
+  contrast: BaseButtonProps['contrast'];
+};
+
+const getColorToken = ({
+  property,
+  variant,
+  state,
+  contrast,
+  intent,
+}: ColorTokenConfig):
+  | `action.${ColorTokenConfig['property']}.${ColorTokenConfig['variant']}.${ColorTokenConfig['state']}`
+  | `feedback.${Required<
+      ColorTokenConfig['intent']
+    >}.action.${ColorTokenConfig['property']}.primary.${ColorTokenConfig['state']}.${Required<
+      ColorTokenConfig['contrast']
+    >}Contrast` => {
+  if (intent && contrast) {
+    return `feedback.${intent}.action.${property}.primary.${state}.${contrast}Contrast`;
```

Comment:
Because for intent, it's always primary. We do not support secondary variants for intents. Even for alert we don't use secondary, its primary only - just low contrast

Reply:
let's discuss this

Reply:
Keeping it hardcoded until we revisit need for other variants here